src/Core/Framework/DataAbstractionLayer/Search/EntitySearchResult.php

```diff
@@ -168,6 +168,18 @@ public function setEntity(string $entity): void
         $this->entity = $entity;
     }
 
+    public function getPageCount(): int
+    {
+        $total = $this->getTotal();
+
+        //next page mode fetches only the next pages with, not the exact count
+        if ($this->getCriteria()->getTotalCountMode() === Criteria::TOTAL_COUNT_MODE_NEXT_PAGES) {
+            $total += $this->getCriteria()->getOffset();
```

Comment:
Is not covered by tests

Reply:
It should be tested by: https://github.com/shopware/platform/pull/2390/files#diff-61eff07120696cdc628dee5f038aebf1fb58ecd30f80f3b2c0a925100648ad33R106-R113

But I moved the test file to the unit test directory (as it only contained unit tests).